### DISCUSSION TITLE
Versioning not showing on page

### DISCUSSION THREAD

**Discussion by tvdboom** [squidfunk/mkdocs-material#3619] [Q&A] [answered]
Hi, I read and followed the documentation regarding versioning and added multiple versions to the documentation page I build. 

Link to repo: [https://github.com/tvdboom/ATOM](https://github.com/tvdboom/ATOM)
Link to docs: [https://tvdboom.github.io/ATOM/](https://tvdboom.github.io/ATOM/)

I encounter multiple problems:

1. The versioning option does not appear on the website. When I try it locally (running `mike serve` it does). What am I doing wrong?
2. I added the Outdated Banner code to `docs_sources/overrides/main.html` but the banner is not showing when I go to an older version (when ran locally). The file looks as follows:
```
{% extends "base.html" %}

{% block htmltitle %}
  <title>{% if page.title and page.title != config.site_name %}{{ page.title }} - {% endif %}{{ config.site_name }}</title>
{% endblock %}

{% block outdated %}
  You're not viewing the latest version.
  <a href="{{ '../' ~ base_url }}">
    <strong>Click here to go to latest.</strong>
  </a>
{% endblock %}
```

3. I have written a custom front page (see `docs_sources/overrides/home.html`). I would like versioning to also appear in the header of this main page (it does not by default?). Is there a quick way to include it?
4. Not related to versioning, but since I am already asking... Is there a way to **not** make the site_name (name of the library) bold in the header? And if that's not possible. How can I make it bold in my custom front page? For some reason those two do not use the same style.

Thanks in advance

**Comment by squidfunk** [comment, marked as answer]
I'm a little short on time, so I can only address 1. and 4. right now.

Regarding 1.: Do you publish your documentation with mike? See:
https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/#publishing-a-new-version

Regarding 4.: Yes, just use [additional CSS](https://squidfunk.github.io/mkdocs-material/customization/#additional-css).

**Reply by tvdboom** [reply]
Hi. Thanks for the reply.

I managed to solve 4 using custom CSS, although, for some reason it only works with `mkdocs serve` but not when I run `mike serve`.

Answer to 1: Yes, I followed the documentation step by step and published using mike. When I run mike list on my terminal I get:
```
4.12 [latest]
4.11
4.10
4.9
4.8
4.7
4.6
4.5
4.4
4.3
4.2
4.1
4.0
```
Also, the `gh-pages` branch on the repo is updated with all versions. Still, the github pages is loading the mkdocs (without mike) version since the URL is still the old one, without the /version/. It seems that github is not "aware" of the mike pages.

Any idea what the problem might be?

**Reply by squidfunk** [reply]
Yes, if you look into the console of your browser, it appears that the `versions.json` is not correctly resolved as the path seems to be incorrect. IMHO this happens because usually, mike will append the version to the URL which Material for MkDocs expects when resolving the `versions.json` file.

For example, when you visit https://squidfunk.github.io/mkdocs-material-example-versioning/, you are immediately redirected to https://squidfunk.github.io/mkdocs-material-example-versioning/0.2/ from which the `versions.json` is correctly resolved. For some reason, this does not happen on your GitHub Pages deployment. I can't say for sure what's causing this, because I don't have the time to check your entire source code, but this is very likely what's causing this.

**Reply by tvdboom** [reply]
The problem was that I was running the page from `master` instead of `gh-pages`. It works now.

Do you have an answer for problem 2 and 3? The outdated warning never shows and I would like to see the versioning also on the front page.

Another minor thing, not sure if caused by mkdocs-material or mike: every first time I load the page, the versioning indicator appears shifted to the top of the screen (image 1). It goes to the "normal" location as soon as I hit refresh or load another page (image 2). It seems like a buggy behaviour.

![temp](https://user-images.githubusercontent.com/32366550/155742297-e2d684ea-9cd6-40eb-9bb9-ad072ea80e86.png)

![temp2](https://user-images.githubusercontent.com/32366550/155742461-8f062877-b067-4ab6-a2da-ec1e3aaa844e.png)

**Reply by squidfunk** [reply]
No, I hadn't had the time to look into them, sorry. I would have to run and debug the code myself. As general advice make sure that your customizations do not interfere with the theme's logic. Since you have overrides in place I would try to debug it by removing stuff one-by-one, checking where it goes wrong. Since the hosted versioning example does not exhibit this behavior, I'm rather sure that it's related to your customizations.

**Reply by squidfunk** [reply]
Oh, and for the outdated banner to work, you need to deploy the outdated banner with that version. The logic is simple: the application will check if the current version is equal to the latest, and then show the banner. If you deployed an earlier version without the banner, it will not work, obviously. `mike` will only deploy the versions you tell it to.

For the first visit, this is executed:

https://github.com/squidfunk/mkdocs-material/blob/4f641cc0ee7befd440956e53b8798899482f6ed4/src/assets/javascripts/integrations/version/index.ts#L113-L130

On subsequent visits, this will trigger the outdated banner:

https://github.com/squidfunk/mkdocs-material/blob/4f641cc0ee7befd440956e53b8798899482f6ed4/src/partials/javascripts/outdated.html#L24-L29

**Reply by tvdboom** [reply]
Thanks for the explanation. I fixed the outdated banner. I'm going to try to debug the other issue now...

**Reply by tvdboom** [reply]
@squidfunk Are you sure the version menu shows when `index.md` is overridden with a html file? I tried removing everything in the html file except the file itself (including custom templates) but it still shows only the header **without** the version menu. When I use an equally empty `index.md` file, the header shows **with** the version menu. In your example page [https://squidfunk.github.io/mkdocs-material-example-versioning/0.2/](https://squidfunk.github.io/mkdocs-material-example-versioning/0.2/), index is a markdown file, not a html file.

**Reply by squidfunk** [reply]
I'm not sure I understand what you mean by "`index.md` is overridden with a HTML file". Nonetheless, versioning does have nothing to do with Markdown content. The application will load the `version.json`, analyze the URL and then display the version selector. It sounds like there's some other error that interrupts the execution of the application, thus the version selector code is never executed. Make sure to check you browser's console.

**Reply by tvdboom** [reply]
I was able to fix it. The problem was indeed in one of the overrides. Thanks again for taking the time to explain how it works. That definitely helped
